src/libsyntax/ast.rs

```diff
@@ -456,7 +472,7 @@ pub enum MetaItemKind {
     /// List meta item.
     ///
     /// E.g. `derive(..)` as in `#[derive(..)]`
-    List(InternedString, Vec<P<MetaItem>>),
+    List(InternedString, Vec<MetaListItem>),
```

Comment:
`P` is owned though, so they wouldn't be shared, unless you mean just moving the value around? Not only that, but ASTs shouldn't get cloned all the time.